accepted/00000-ptf-repos-in-uyuni.md

```diff
@@ -0,0 +1,104 @@
+The defined path is:
+
+    PTF/Release/<Customer Account ID>/<Product Identifier>/<Product Version>/<Product Architecture>/ptf[_debug]/
+
+When Uyuni is listing the repositories available for an organizations, it may find repositories which `id` is not known in the complete product tree.
+To identify them as PTF repositories we can check if the path start with `PTF/Release`.
+
+If they are PTF repositories, we need to generate the missing values for the Channel mapping like this:
+
+- repository id: take it from the repository definition
+- product id: search for the product using `Product Identifier`, `Product Version`, and `Product Architecture` like specified in the URL
+- root product id: find all existing root products for the product id. We need to generate an entry for all of them
+
+The values for update tag and mandatory can be hard coded as they are the same 
+for all PTF repositories.
+
+- update tag: empty
+- mandatory: false
+
+As we need to generate entries for all existing root products we can iterate over all we found.
+For every root product we take the first channel we can find which does have a `parent channel label` set.
+We can take this as parent channel label for our new entry.
+
+As the channel label and name must be unique.
+We can generate the label and name from the URL components:
+
+    <Customer Account ID> <Product Identifier> <Product Version> "PTFs" ["Debuginfo"] <Product Architecture>
+
+The label is computed using the lowercase components joined with a dash ("-") and the name just joined with a space (" ").
+
+This can either be done by Uyuni Server or SCC. In case SCC use this algorithm it could set it as `repository name`
+joined with a dash ("-") and in `repository description` using spaces.
+
+The product tree uses additional suffixes for products which belongs to multiple root products to make label and name unique.
+The suffix is appended always after the architecture of the channel label.
+We can parse it from the entry we used to find the `parent channel label` and use it to enhance the component list.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- we are guessing the values. Some, like the channel label, are sensitive and must not be changed. We have to be sure that the rules and the structure never changes.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Stop using product unscoped SCC endpoint and use the product scope of the requesting organization. In this endpoint we see the PTF repositories directly with its ids.
+SCC needs to be adapted to provide the channel labels and names directly in its API.
+
+Downside:
+
+- when multiple organization credentials are used, we need to request and parse multiple product scopes which cost much more time.
+- we might need to do bigger changes in the backend as we need to adapt to getting only parts of the data.
+- dependency to other product development teams which slow down the development
```

Comment:
```suggestion
- in Public Clouds, PAYG products might be used - which means products the user has no SCC subscription for. Uyuni might still need product data about those
```

Reply:
Not sure if PAYG customers can get PTFs at all. They do not have a contract with SUSE.

Reply:
FTR: I checked and right, at the moment this is not possible.